codegen-server/typescript/src/main/kotlin/software/amazon/smithy/rust/codegen/server/typescript/smithy/generators/TsApplicationGenerator.kt

```diff
@@ -0,0 +1,303 @@
+                )
+            }
+        }
+        Attribute("""napi(object)""").render(writer)
+        writer.rustBlock("pub struct TsHandlers") {
+            operations.map { operation ->
+                val operationName = symbolProvider.toSymbol(operation).name
+                val input = "${operationName}Input"
+                val output = "${operationName}Output"
+                val fnName = operationName.toSnakeCase()
+                rustTemplate(
+                    """
+                    ##[napi(ts_type = "(input: $input) => Promise<$output>")]
+                    pub $fnName: #{napi}::JsFunction,
+                    """,
+                    *codegenScope,
+                )
+            }
+        }
+    }
+
+    private fun renderApp(writer: RustWriter) {
+        Attribute("napi").render(writer)
+        writer.rust(
+            """
+            pub struct App {
+                handlers: Handlers,
+            }
+            """,
+        )
+        Attribute("napi").render(writer)
+        writer.rustBlock("impl App") {
+            renderAppCreate(writer)
+            renderAppStart(writer)
+        }
+    }
+
+    private fun renderAppCreate(writer: RustWriter) {
+        Attribute("napi(constructor)").render(writer)
+        writer.rustBlockTemplate(
+            """pub fn create(ts_handlers: TsHandlers) -> #{napi}::Result<Self>""",
+            *codegenScope,
+        ) {
+            operations.map { operation ->
+                val operationName = symbolProvider.toSymbol(operation).name
+                val input = "crate::input::${operationName}Input"
+                val fnName = operationName.toSnakeCase()
+                rustTemplate(
+                    """
+                    let $fnName: #{napi}::threadsafe_function::ThreadsafeFunction<
+                        $input, #{napi}::threadsafe_function::ErrorStrategy::Fatal
+                    > = ts_handlers.$fnName.create_threadsafe_function(0, |ctx| Ok(vec![ctx.value]))?;
+                    """,
+                    *codegenScope,
+                )
+            }
+            rust("let handlers = Handlers {")
+            operations.map { operation ->
+                val operationName = symbolProvider.toSymbol(operation).name.toSnakeCase()
+                rust("    $operationName: $operationName.clone(),")
+            }
+            rust("};")
+            writer.rust("Ok(Self{ handlers })")
+        }
+    }
+
+    private fun renderAppStart(writer: RustWriter) {
+        Attribute("napi").render(writer)
+        writer.rustBlockTemplate(
+            """pub fn start(&self, socket: &TsSocket) -> #{napi}::Result<()>""",
+            *codegenScope,
+        ) {
+            rustTemplate(
+                """
+                // #{SmithyTs}::setup_tracing();
+                let plugins = #{SmithyServer}::plugin::PluginPipeline::new();
+                let builder = crate::service::PokemonService::builder_with_plugins(plugins);
```

Comment:
This have to change to that we use the proper service name coming from the smithy model we are generating for instead of the hardcoding the PokemonService.